iteration/iteration.go

```diff
@@ -69,6 +75,7 @@ type Repository interface {
 	CanStart(ctx context.Context, i *Iteration) (bool, error)
 	LoadMultiple(ctx context.Context, ids []uuid.UUID) ([]Iteration, error)
 	LoadChildren(ctx context.Context, parentIterationID uuid.UUID) ([]Iteration, error)
+	CalculateActiveIterationOnUpdate(ctx context.Context, itr *Iteration) (*bool, error)
```

Comment:
right.. but.. 

* Why does CalculateActiveIterationOnUpdate need Database access to compare some dates?
* CalculateActiveIterationOnUpdate is called from within Save
* * Why is CalculateActiveIterationOnUpdate exposed on the Repository?
* * Why does CalculateActiveIterationOnUpdate reload the Iteration from the database, it's already loaded by Save?